lib/state/state.js

```diff
@@ -517,36 +482,62 @@ State.prototype.respond = function (response, isJson) {
 	this.response = isJson ? response : JSON.stringify(response);
 };
 
+/**
+ * Distribute all events currently stacked on
+ * this state object.
+ *
+ * If you wish to also distribute archivist mutations stacked
+ * on this state, please have a look at the `distribute` method
+ * instead.
+ *
+ * @param {Function} cb callback function
+ */
+State.prototype.distributeEvents = function (cb) {
+	this.otherEvents[this.actorId] = this.myEvents.slice();
```

Comment:
It's possible for `this.actorId` to not be set. You'll probably want to do something there.

But the only 2 things I can currently think of are:

- throw an exception / log an error
- ignore `myEvents`

Neither is very appealing, and we can probably do better by preventing this entire situation from arising by throwing at emit-time. That is, you may emit to `undefined / null / true` if that means emitting to `state.actorId` (as is already the case), but that will throw if `state.actorId` is not set.

Then a check here that either logs or skips `myEvents` if `this.actorId` is not set would become a rarer edge-case (implying the user is really messing about with the system in ways they shouldn't be doing).

Reply:
`myEvents` can only be empty if there are not `actorId` set, so ignoring would make sense I think.